### PR TITLE
Add circleCI

### PR DESCRIPTION
**Pull request by brendanburkhart**

### CODE REVIEW

**Comment by fharding1** [comment]
Not DockerHub Auto Test? smh

:)

**Comment by brendanburkhart** [comment]
Should I actually use that? CircleCI is relatively fast but other than that I'm not particularly smitten with it.

**Comment by fharding1** [comment]
Docker Hub Auto Test can be super slow but otherwise I like it a lot more than CircleCI. If CircleCI does the job :man_shrugging:

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+      - image: gradle:5.4.1-jdk11-slim
+    environment:
+      JVM_OPTS: -Xmx3200m
+    steps:
+      - checkout
+      - restore_cache:
+          key: jars-{{ checksum "build.gradle" }}
+      - run:
+          name: Chmod permissions
+          command: chmod +x ./gradlew
```
Comment (fharding1) [inline .circleci/config.yml]: You should probably just `chmod` the file in the repo and remove this step. Git should retain the permissions.

**Pull request merged**